pkg/cloudproviders/gcp/cred_manager.go

```diff
@@ -0,0 +1,30 @@
+package gcp
+
+import (
+	"context"
+
+	"golang.org/x/oauth2/google"
+)
+
+// CredentialsManager manages GCP credentials based on the environment.
+type CredentialsManager interface {
+	Start() error
+	Stop()
+	GetCredentials(ctx context.Context) (*google.Credentials, error)
+}
+
+// DefaultCredentialsManager always returns the default GCP credential chain.
+type DefaultCredentialsManager struct{}
```

Comment:
Sorry can you explain what you mean? I created this struct as a fallback when the creation of a regular `gcpCredentialsManagerImpl` fails. This can happen when the K8s client fails, in which case rather than panicking I thought we might want to continue the main loop without crashing anything. How would this work with a function?

Reply:
Ah I see thanks. This is refactored in the next PR such that the struct is no longer exposed.